### ISSUE TITLE
[Bug] Greedy optimisation cuts some subpaths in wrong direction

### ISSUE THREAD

**Issue by Sophist-UK** [meerk40t/meerk40t#1028]
v0.7.6

As you can see in the attached screenshot some leather stitch holes are burned correctly and some are burned with each quarter circle burned in the incorrect direction. Some of these holes were mirrored in Inkscape so path direction is opposite.

![image](https://user-images.githubusercontent.com/3001893/166718460-557f3bbc-ef5f-462d-9d87-8135bf7a3579.png)

In the picture above, the incorrect holes are anti-clockwise, and the correct ones clockwise, however there are anti-clockwise holes elsewhere which are burned correctly.

As you can see in the following it appears that the travel is done to the incorrect point of the chosen segment, and it goes wrong from there.

![image](https://user-images.githubusercontent.com/3001893/166721654-26b59f77-84ee-4087-9b29-8d094aebd09a.png)

**Comment by Sophist-UK** [comment]
I think I have tracked this down to a codacy fix I introduced in #795 where I moved some code into a sub-method `short_travel_cutcode_candidate`.

`short_travel_cutcode_candidate` needs to return not just `closest` but also backwards.

So [line 939](https://github.com/meerk40t/meerk40t/blob/legacy7/meerk40t/core/cutplan.py#L939) should read:
```python
                closest, backwards = self.short_travel_cutcode_candidate(
                    context, closest, curr, distance
                )
```
and [line 1029](https://github.com/meerk40t/meerk40t/blob/legacy7/meerk40t/core/cutplan.py#L1029) should read:
```python
        return closest, backwards
```

**Comment by Sophist-UK** [comment]
```
MeerK40t crash log. Version: 0.7.6002 on Windows:AMD64 - 4.1.1 msw (phoenix) wxWidgets 3.1.5
Traceback (most recent call last):
  File "meerk40t\gui\wxmribbon.py", line 215, in open_simulator
  File "meerk40t\kernel.py", line 197, in __call__
  File "meerk40t\kernel.py", line 1956, in console
  File "meerk40t\kernel.py", line 2042, in _console_parse
  File "meerk40t\kernel.py", line 1378, in inner
  File "meerk40t\core\planner.py", line 677, in plan_optimize
  File "meerk40t\core\cutplan.py", line 74, in execute
  File "meerk40t\core\cutplan.py", line 345, in optimize_travel
  File "meerk40t\core\cutplan.py", line 940, in short_travel_cutcode
  File "meerk40t\core\cutplan.py", line 1029, in short_travel_cutcode_candidate
UnboundLocalError: local variable 'backwards' referenced before assignment
```
